pkg/services/secrets/kvstore/plugin_fatal_crash_test.go

```diff
@@ -22,44 +21,42 @@ import (
 // Set fatal flag to true, then simulate a plugin start failure
 // Should result in an error from the secret store provider
 func TestFatalPluginErr_PluginFailsToStartWithFatalFlagSet(t *testing.T) {
-	svc, mgr, _, _, err := setupFatalCrashTest(t, true, true, false)
-	_ = fmt.Sprint(mgr) // this is here to satisfy the linter
+	p, err := setupFatalCrashTest(t, true, true, false)
 	require.Error(t, err)
-	require.Nil(t, svc)
+	require.Nil(t, p.secretsKVStore)
 }
 
 // Set fatal flag to false, then simulate a plugin start failure
 // Should result in the secret store provider returning the sql impl
 func TestFatalPluginErr_PluginFailsToStartWithFatalFlagNotSet(t *testing.T) {
-	svc, mgr, _, _, err := setupFatalCrashTest(t, true, false, false)
-	_ = fmt.Sprint(mgr) // this is here to satisfy the linter
+	p, err := setupFatalCrashTest(t, true, false, false)
 	require.NoError(t, err)
-	require.IsType(t, &CachedKVStore{}, svc)
-	cachedKv, _ := svc.(*CachedKVStore)
+	require.IsType(t, &CachedKVStore{}, p.secretsKVStore)
+	cachedKv, _ := p.secretsKVStore.(*CachedKVStore)
 	require.IsType(t, &secretsKVStoreSQL{}, cachedKv.GetUnwrappedStore())
 }
 
 // With fatal flag not set, store a secret in the plugin while backwards compatibility is disabled
 // Should result in the fatal flag going from unset -> set to true
 func TestFatalPluginErr_FatalFlagGetsSetWithBackwardsCompatDisabled(t *testing.T) {
-	svc, _, kvstore, _, err := setupFatalCrashTest(t, false, false, true)
+	p, err := setupFatalCrashTest(t, false, false, true)
 	require.NoError(t, err)
-	require.NotNil(t, svc)
-	err = svc.Set(context.Background(), 0, "datasource", "postgres", "my secret")
+	require.NotNil(t, p.secretsKVStore)
+	err = p.secretsKVStore.Set(context.Background(), 0, "datasource", "postgres", "my secret")
 	require.NoError(t, err)
-	isFatal, err := isPluginStartupErrorFatal(context.Background(), GetNamespacedKVStore(kvstore))
+	isFatal, err := isPluginStartupErrorFatal(context.Background(), GetNamespacedKVStore(p.kvstore))
 	require.NoError(t, err)
 	require.True(t, isFatal)
 }
```

Comment:
nit: I like to separate the code into logic units (Method calls with assertions) to make it more readable.

Reply:
you could also just separate them with an empty between setup, execution and validation of the test

Reply:
fixed here https://github.com/grafana/grafana/pull/54231/commits/acb3abdec06bd896a75c3280ad299269958129bb